src/lib.rs

```diff
@@ -1,6 +1,13 @@
+#![allow(clippy::too_many_arguments)]
+// We are writing to binary files, so there are cases where we specifically need
+// `\n` and not the platform-specific newline character.
+#![allow(clippy::write_with_newline)]
```

Comment:
writeln will always use `\n`.

https://doc.rust-lang.org/stable/std/macro.writeln.html

> On all platforms, the newline is the LINE FEED character (\n/U+000A) alone (no additional CARRIAGE RETURN (\r/U+000D).

Reply:
Ah, good to know!
I've switched to using `writeln` where clippy recommends, although I feel that the explicit `\n` might be better - thoughts?

Reply:
Still working on reviewing, but from what I have seen, yeah an explicit `\n` is probably nicer.

Reply:
I switch back to explicit `\n`